src/components/ProductDetail.tsx

```diff
@@ -0,0 +1,63 @@
+import React from 'react';
+import Loading from './Loading';
+import Button from "./Button"
+import { useState } from 'react';
+import { useFetch } from '../Functions/useFetch';
+import { usePost } from '../Functions/usePost';
+import { useParams, useNavigate } from 'react-router-dom';
+import { useDispatch, useSelector } from 'react-redux'
+import { AddedToCart } from './../redux/cartSlice'
+
+const ProductDetail = () => {
+    const [userId, setUserId] = useState('')
+
+    const productID = Number(useParams().productID) || 0;
+
+    const {data: product, fetchLoading, fetchError} = useFetch('products/' + productID);
+
+    const dispatch = useDispatch()
+    const users = useSelector(state => state.users)
+
+    const navigate = useNavigate();
+    const handleShowProducts = () => {
+        navigate('/products');
+    };
+
+    const handleAddToCart = (productID, productTitle) => {
+        dispatch(AddedToCart('1', productID, productTitle))
+    }
+    
+    if(fetchError != undefined && fetchError.length > 0) { return <p style={{ textAlign:'center'}}>An error occurres {fetchError}</p>; };
+
+    if (fetchLoading) { return <Loading /> }
```

Comment:
No brackets needed if you use one line.
`if (fetchLoading) return <Loading /> `